ui/app/AppLayouts/Profile/helpers/VisibilityAndPositionDirtyStateModel.qml

```diff
@@ -49,18 +49,25 @@ WritableProxyModel {
             return
 
         // hiding, changing visibility level
-        if (visibility === visibilityHidden
-                || oldVisibility !== visibilityHidden) {
-            set(sourceIdx, { showcaseVisibility: visibility })
+        if (visibility === visibilityHidden) {
+            set(sourceIdx, { showcaseVisibility: undefined, showcasePosition: undefined})
```

Comment:
It's a corner case because we're saving only the visible showcase items. As a result we won't have any item with `NoOne` in the source model. `Constants.ShowcaseVisibility.NoOne` is just a UI value. When the backend model changes after save, all hidden items coming from the backend will have `undefined` as `showcaseVisibility` and `showcasePosition`. These values needs to match in the writable in order to drop the `dirty` flag